Add support for Travis CI

tuliom opened the pull request:
Test on Ubuntu 20.04, using GCC 8, 9, AT 14.0 (GCC 10) and clang 10.

Notice this patch alone will not enable Travis CI on this project.
It also requires the owner of the repository to enable the tests on
https://travis-ci.org/github/open-power-sdk/pveclib

Signed-off-by: Tulio Magno Quites Machado Filho <tuliom@linux.ibm.com>

tuliom commented:
An example of test is available at https://travis-ci.org/github/tuliom/pveclib/builds/745965985

munroesj52 reviewed (changes requested):
Can you add Clang-8/9/10 to this? Clang 8 seems to be the minimum for any reasonable code generation.

munroesj52 commented:
Was planning to look at this next (after gather/scatter pass 2 commit).

tuliom commented:
@munroesj52 I updated this PR and added clang 8 and 9.
Sample output is available at: https://travis-ci.org/github/tuliom/pveclib/builds/746073641

> Was planning to look at this next (after gather/scatter pass 2 commit).

No problem.  This PR can wait.

tuliom commented:
For the record, all the tests passed after creating a branch with all the PRs open right now:
https://travis-ci.org/github/tuliom/pveclib/builds/746083344

munroesj52 commented:
tuliom: "For the record, all the tests passed after creating a branch with all the PRs open right now:"

Excellent Thanks

munroesj52 commented:
I logged in and authorized Travis CI.
Looks like all tests ran successfully

Thanks

On Wed, Nov 25, 2020 at 3:46 PM Tulio Magno Quites Machado Filho <
notifications@github.com> wrote:

> Test on Ubuntu 20.04, using GCC 8, 9, AT 14.0 (GCC 10) and clang 10.
>
> Notice this patch alone will not enable Travis CI on this project.
> It also requires the owner of the repository to enable the tests on
> https://travis-ci.org/github/open-power-sdk/pveclib
>
> Signed-off-by: Tulio Magno Quites Machado Filho tuliom@linux.ibm.com
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/open-power-sdk/pveclib/pull/128
> Commit Summary
>
>    - Add support for Travis CI
>
> File Changes
>
>    - *A* .travis.yml
>    <https://github.com/open-power-sdk/pveclib/pull/128/files#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485>
>    (48)
>    - *M* README.md
>    <https://github.com/open-power-sdk/pveclib/pull/128/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5>
>    (2)
>
> Patch Links:
>
>    - https://github.com/open-power-sdk/pveclib/pull/128.patch
>    - https://github.com/open-power-sdk/pveclib/pull/128.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/open-power-sdk/pveclib/pull/128>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIEI2IPPGUASZEYPIF4VCYTSRV3LRANCNFSM4UC5YFLA>
> .
>

munroesj52 commented:
I have seen occasional failure in Travis CI checking for updates for AT
releases.

Err:6 https://public.dhe.ibm.com/software/server/POWER/Linux/toolchain/at/ubuntu
focal Release

  Could not wait for server fd - select (11: Resource temporarily
unavailable) [IP: 170.225.15.112 443]

Reading package lists...

E: The repository
'https://public.dhe.ibm.com/software/server/POWER/Linux/toolchain/at/ubuntu
focal Release' does not have a Release file.

The command "sudo apt-get update" failed and exited with 100 during .

Is this a common problem or should I just retry?


On Wed, Nov 25, 2020 at 3:46 PM Tulio Magno Quites Machado Filho <
notifications@github.com> wrote:

> Test on Ubuntu 20.04, using GCC 8, 9, AT 14.0 (GCC 10) and clang 10.
>
> Notice this patch alone will not enable Travis CI on this project.
> It also requires the owner of the repository to enable the tests on
> https://travis-ci.org/github/open-power-sdk/pveclib
>
> Signed-off-by: Tulio Magno Quites Machado Filho tuliom@linux.ibm.com
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/open-power-sdk/pveclib/pull/128
> Commit Summary
>
>    - Add support for Travis CI
>
> File Changes
>
>    - *A* .travis.yml
>    <https://github.com/open-power-sdk/pveclib/pull/128/files#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485>
>    (48)
>    - *M* README.md
>    <https://github.com/open-power-sdk/pveclib/pull/128/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5>
>    (2)
>
> Patch Links:
>
>    - https://github.com/open-power-sdk/pveclib/pull/128.patch
>    - https://github.com/open-power-sdk/pveclib/pull/128.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/open-power-sdk/pveclib/pull/128>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIEI2IPPGUASZEYPIF4VCYTSRV3LRANCNFSM4UC5YFLA>
> .
>

tuliom commented:
I've never seen that. I don't know what to do.

munroesj52 commented:
Looks like a timing/connectivity issue. Retried and it ran.

On Wed, Dec 16, 2020 at 8:06 PM Tulio Magno Quites Machado Filho <
notifications@github.com> wrote:

> I've never seen that. I don't know what to do.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/open-power-sdk/pveclib/pull/128#issuecomment-747156655>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIEI2IIKVXX3QY5JLFOZPN3SVFRTDANCNFSM4UC5YFLA>
> .
>
